pkg/controller/scyllacluster/resource.go

```diff
@@ -219,9 +221,25 @@ func StatefulSetForRack(r scyllav1.RackSpec, c *scyllav1.ScyllaCluster, existing
 	if err != nil {
 		return nil, fmt.Errorf("can't get rack labels: %w", err)
 	}
-	matchLabels := helpers.ShallowCopyMap(rackLabels)
```

Comment:
nit: Should we also remove the ShallowCopyMap function with its only usage gone?

Reply:
done